src/pages/klage/vurderFormkrav/VurderFormkrav.tsx

```diff
@@ -0,0 +1,271 @@
+    const { formatMessage } = useI18n({ messages });
+    const [lagreStatus, lagre] = useAsyncActionCreator(klageActions.vurderFormkrav);
+    const [bekreftStatus, bekreft] = useAsyncActionCreator(klageActions.bekreftFormkrav);
+
+    const initialValues = {
+        vedtakId: props.klage.vedtakId,
+        innenforFristen: props.klage.innenforFristen,
+        klagesDetPåKonkreteElementerIVedtaket: props.klage.klagesDetPåKonkreteElementerIVedtaket,
+        signert: props.klage.erUnderskrevet,
+        begrunnelse: props.klage.begrunnelse,
+    };
+
+    const {
+        handleSubmit,
+        control,
+        reset,
+        formState: { isDirty, isSubmitSuccessful },
+    } = useForm<FormData>({
+        defaultValues: initialValues,
+    });
+
+    const handleLagreFormkrav = (values: FormData) => {
+        if (eqFormData.equals(values, initialValues)) {
+            return;
+        }
+
+        lagre(
+            {
+                sakId: props.sakId,
+                klageId: props.klage.id,
+                vedtakId: values.vedtakId,
+                innenforFristen: values.innenforFristen,
+                klagesDetPåKonkreteElementerIVedtaket: values.klagesDetPåKonkreteElementerIVedtaket,
+                erUnderskrevet: values.signert,
+                begrunnelse: values.begrunnelse,
+            },
+            (klage) => {
+                //vi resetter formet, slik at tilstandssjekken for å bekrefte og fortsette har de nye dataene
+                reset({
+                    vedtakId: klage.vedtakId,
+                    innenforFristen: klage.innenforFristen,
+                    klagesDetPåKonkreteElementerIVedtaket: klage.klagesDetPåKonkreteElementerIVedtaket,
+                    signert: klage.erUnderskrevet,
+                    begrunnelse: klage.begrunnelse,
+                });
+            }
+        );
+    };
+
+    const iGyldigTilstandForÅBekrefteOgFortsette = () => {
+        return !erKlageVilkårsvurdertUtfyltEllerSenere(props.klage) || (isDirty && !isSubmitSuccessful);
+    };
+
+    const handleBekreftOgFortsettClick = () => {
+        if (erKlageVilkårsvurdertBekreftetEllerSenere(props.klage) && !isDirty) {
+            history.push(
+                Routes.klage.createURL({
+                    sakId: props.sakId,
+                    klageId: props.klage.id,
+                    steg: KlageSteg.Vurdering,
+                })
+            );
+            return;
+        }
+        bekreft(
+            {
+                sakId: props.sakId,
+                klageId: props.klage.id,
+            },
+            () => {
+                history.push(
+                    Routes.klage.createURL({
+                        sakId: props.sakId,
+                        klageId: props.klage.id,
+                        steg: KlageSteg.Vurdering,
+                    })
+                );
+            }
+        );
+    };
+
+    if (!iGyldigTilstandForÅVilkårsvurdere(props.klage)) {
```

Comment:
Føles litt rart at man må lagre før man kan bekrefte og gå videre. Men kanskje en god tanke bak? I tillegg mangler det litt feedback på validering her og der, så når man trykker `lagre` på en invalid-form så skjer det ingenting

Reply:
Taken er at man skal ha mulighet til å lagre en ugyldig form. Slik at man kan jobbe, og fortsette en annen gang uten at vi skal tvinge dem å fylle ut alt (som vi gjør ved søknadsbehandling og revurdering? sin lagre og fortsett senere). 

Siden man ikke hadde mulighet til å bekrefte og fortsette senere i feil tilstand, var den eneste muligheten for saksbehandler å bare lagre behandlingen. Da var det ikke nødvendig med noe input validering frontend. Vi gjemte vekk bare bekreft og fortsett knappen. Derimot, nå, så viser vi knappen hele tiden, så det er lagt inn validering når man bekrefter, og en custom melding om at man må 'lagre, og alt må være utfyllt'-ish før man kan fortsette.